File: ios/iLoppis/UI/Theme/AppColors.swift
import SwiftUI

/// Central color palette mirroring Android AppColors.
struct AppColors {
    // Brand colors
    static let background = Color(red: 0.960, green: 0.910, blue: 0.913) // #F5E8E9
    static let cardBackground = Color(red: 0.980, green: 0.953, blue: 0.957) // #FAF3F4
    static let dialogBackground = Color.white
    static let primary = Color(red: 0.718, green: 0.110, blue: 0.110) // #B71C1C
    static let primaryLight = Color(red: 0.875, green: 0.349, blue: 0.192) // #DF5931
    static let border = Color(red: 0.886, green: 0.909, blue: 0.941) // #E2E8F0

    // Text colors
    static let textPrimary = Color(red: 0.176, green: 0.216, blue: 0.282) // #2D3748
    static let textSecondary = Color(red: 0.290, green: 0.333, blue: 0.408) // #4A5568
    static let textMuted = Color.gray
    static let textDark = Color(.darkGray)
    static let textError = Color(red: 0.898, green: 0.243, blue: 0.243) // #E53E3E

    // Button semantic colors
    static let buttonPrimary = primary
    static let buttonPrimaryDisabled = Color(red: 0.796, green: 0.835, blue: 0.878) // #CBD5E0
    static let buttonSecondary = Color(red: 0.443, green: 0.502, blue: 0.588) // #718096

    // Status colors
    static let success = Color(red: 0.298, green: 0.686, blue: 0.314) // #4CAF50
    static let warning = Color(red: 0.965, green: 0.678, blue: 0.333) // #F6AD55
    static let error = Color(red: 0.898, green: 0.243, blue: 0.243) // #E53E3E
    static let info = Color(red: 0.129, green: 0.588, blue: 0.953) // #2196F3

    // Input colors
    static let inputBackground = Color(red: 0.969, green: 0.980, blue: 0.988) // #F7FAFC
    static let inputBorder = border
    static let inputBorderFocused = buttonSecondary

    // Badge colors
    static let badgeOpenBackground = Color(red: 0.784, green: 0.902, blue: 0.792) // #C8E6C9
    static let badgeOpenText = Color(red: 0.220, green: 0.557, blue: 0.235) // #388E3C
    static let badgeUpcomingBackground = Color(red: 0.910, green: 0.784, blue: 0.792) // #E8C8CA
    static let badgeUpcomingText = Color(red: 0.427, green: 0.298, blue: 0.255) // #6D4C41
    static let badgeDefaultBackground = Color(red: 0.878, green: 0.878, blue: 0.878) // #E0E0E0
    static let badgeDefaultText = Color(red: 0.459, green: 0.459, blue: 0.459) // #757575

    // Legacy aliases
    static let buttonSuccess = success
    static let buttonInfo = info
    static let buttonDanger = error
    static let swishBlue = Color(red: 0.0, green: 0.478, blue: 0.800) // #007ACC
}

File: android/app/src/main/java/se/iloppis/app/ui/theme/AppColors.kt
package se.iloppis.app.ui.theme

import androidx.compose.ui.graphics.Color

/**
 * Central color palette for the iLoppis app.
 * All colors used in the app should be referenced from here.
 * 
 * Button Color Guidelines:
 * - Primary (Orange): Main actions (Scan, Verify, Submit, Continue)
 * - Secondary (Gray): Less important actions (Back, Cancel as button)
 * - TextSecondary: Dismiss/Cancel as text link
 * - Success (Green): Positive confirmations (in result sheets only)
 * - Warning (Amber): Caution states (duplicate, offline)
 * - Error (Red): Error states and destructive actions
 */
object AppColors {
    // Skog (Woods) theme - source: iloppis/frontend/src/styles/themes.js (Themes.Woods)
    val Background = Color(0xFFF9FAFB)
    val CardBackground = Color(0xFFFFFFFF)
    val DialogBackground = Color(0xFFFFFFFF)
    val Primary = Color(0xFF2F6A6A)
    val PrimaryLight = Color(0xFF346E79)  // Woods accent
    val Border = Color(0xFFE5E7EB)

    // Text colors
    val TextPrimary = Color(0xFF132A2F)
    val TextSecondary = Color(0xFF5B6B7A)
    val TextMuted = Color(0xFF5B6B7A)
    val TextDark = Color(0xFF132A2F)
    val TextError = Color(0xFFD32F2F)

    // Button semantic colors
    val ButtonPrimary = Primary  // Main action buttons
    val ButtonPrimaryDisabled = Color(0xFFCBD5E0)
    val ButtonSecondary = Border  // Secondary actions
    
    // Status colors (for result sheets, badges, feedback)
    val Success = Color(0xFF2E7D6B)  // Woods success
    val Warning = Color(0xFFDAA000)  // Woods warning
    val Error = Color(0xFFD32F2F)    // Woods danger
    val Info = Color(0xFF346E79)     // Woods accent

    // Legacy aliases (for existing code compatibility)
    val ButtonSuccess = Success
    val ButtonInfo = Info
    val ButtonDanger = Error
    val SwishBlue = Color(0xFF007ACC)  // Swish brand color

    // Input colors
    val InputBackground = Background
    val InputBorder = Border
    val InputBorderFocused = Primary

    // Badge colors - Open state
    val BadgeOpenBackground = Success.copy(alpha = 0.18f)
    val BadgeOpenText = Success

    // Badge colors - Upcoming state
    val BadgeUpcomingBackground = Warning.copy(alpha = 0.18f)
    val BadgeUpcomingText = Warning

    // Badge colors - Default/Closed state
    val BadgeDefaultBackground = Border
    val BadgeDefaultText = TextSecondary

    // Badge colors - Ongoing (Pågående) state - same as Open
    val BadgeOngoingBackground = BadgeOpenBackground
    val BadgeOngoingText = BadgeOpenText

    // Badge colors - Info/Upcoming (Kommande) state
    val BadgeInfoBackground = Info.copy(alpha = 0.18f)
    val BadgeInfoText = Info

    // Splash screen
    val SplashBackground = Success
    val OnSplashBackground = Color.White

    // On-button text colors
    val OnButtonPrimary = Color.White
    val OnButtonSecondary = TextPrimary

    // Accent / decoration
    val Gold = Warning

    // Surface variants (for containers, backgrounds)
    val SurfaceVariant = Border
    val ErrorContainer = Error.copy(alpha = 0.12f)
    val OnErrorContainer = Error
    val WarningContainer = Warning.copy(alpha = 0.12f)
    val OnWarningContainer = Warning

    // Top-level text on surface
    val OnBackground = TextPrimary
    val NavigatorOverlay = TextPrimary  // onSurface equivalent for scrim
}
